sa/sa.go

```diff
@@ -1375,11 +1378,38 @@ func (ssa *SQLStorageAuthority) NewAuthorizations2(ctx context.Context, req *sap
 		if err != nil {
 			return nil, err
 		}
-		err = ssa.dbMap.Insert(am)
+
+		// Each authz needs a (?,?...), in the VALUES block
+		qs.WriteString(fmt.Sprintf("(%s),", strings.TrimRight(strings.Repeat(" ?,", 11), ",")))
+		// Must follow the order of authzFields
+		vals = append(vals, am.ID)
+		vals = append(vals, am.IdentifierType)
+		vals = append(vals, am.IdentifierValue)
+		vals = append(vals, am.RegistrationID)
+		vals = append(vals, am.Status)
+		vals = append(vals, am.Expires)
+		vals = append(vals, am.Challenges)
+		vals = append(vals, am.Attempted)
+		// AttemptedAt is not included in the authzFields
+		vals = append(vals, am.Token)
+		vals = append(vals, am.ValidationError)
+		vals = append(vals, am.ValidationRecord)
+	}
+	// The VALUES block question-string has a trailing comma, remove it
```

Comment:
I need some help understanding this comment.

Reply:
```
	// At this point, the VALUES block question-string has a trailing comma, we need
	// to remove it to make sure we're valid SQL.
```